### PR TITLE
Collect DisplayVersion to detect new vulnerabilities

### PR DESCRIPTION
**Pull request by MarcelKemp**
|Related issue|
|---|
|#8967|

## Description
Hi team,

With this PR, we have obtained the `DisplayVersion` field to obtain the version reflected in the new _Windows_, in addition to fixing a problem in _Windows 10_ that incorrectly displayed the necessary patch information to fix the reported vulnerability.

To do this, the following steps have been carried out:
- First the new `DisplayVersion` field has been collected with _syscollector_.

- Then it has been used in vulnerability detector to match with its corresponding **_CPE_**.

- And finally, the query that compares the versions of _Windows 10_ has been modified, both to avoid erroneously reporting the patch that fixes the vulnerability and to ensure that the matching of the new `DisplayVersion` with its corresponding patches works correctly.
- In addition, support has been added to `Windows Server 2022`.

## Logs/Alerts example

Agent information:
```
sqlite> select * from sys_osinfo;
0|2021/09/20 11:33:16|WIN10|i686|Microsoft Windows 10 Home N|10.0.19042||10|0||19042|||||2009|1632137595599520900|20H2

sqlite> select * from sys_hotfixes;
0|2021/09/20 10:33:16|KB4562830|c651fa941c06383364d7d47f8d7046499cc099fb
0|2021/09/20 10:33:16|KB4580325|971b360361a480d5a5be94a472dd8cf2d01ef4c0
0|2021/09/20 10:33:16|KB5003503|b608b0991ede7626e1d84473598147d98053380d
0|2021/09/20 10:33:17|KB5003537|53ef92671e0cc7cce60707f6450a05786d3b18ee
0|2021/09/20 10:33:17|KB5003742|9b1d8e0f90436d583371d27f46d9bf69c459f897
0|2021/09/20 10:33:17|KB5004237|cbb721f428a8cc286390db8527395bbca6308ebf
0|2021/09/20 10:33:17|KB5004331|222f14c3216c9ba03c0d986818bec6edad64d16a
0|2021/09/20 10:33:17|KB5005033|93494b3e0283ab397c1a582628a321bd948cb41f
0|2021/09/20 10:33:17|KB5005260|d9fbc6a677f5fcbfd1bcf017d93f73c3cf991533
```

`ossec.log` file:
```
2021/09/20 12:12:57 wazuh-modulesd:vulnerability-detector[114398] wm_vuln_detector.c:5767 at wm_vuldet_insert_agent_data(): DEBUG: (5446): The CPE 'o:microsoft:windows_10:20h2:::::::' from the agent '097' was indexed.
...
2021/09/20 12:18:01 wazuh-modulesd:vulnerability-detector[114398] wm_vuln_detector.c:1535 at wm_vuldet_send_cve_report(): DEBUG: (5467): Agent '097' is vulnerable to 'CVE-2021-1678'. Condition: 'KB5005565 patch is not installed'
```

Alert of a vulnerability:
```
{"timestamp":"2021-09-20T10:33:24.386+0000","rule":{"level":10,
"description":"CVE-2021-1678 affects Windows 10","id":"23505","firedtimes":65,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},
"agent":{"id":"097","name":"win10","ip":"10.0.2.15"},"manager":{"name":"6f790cc7af58"},"id":"1632134004.13465092","decoder":{"name":"json"},"data":{
"vulnerability":{"package":{"name":"Windows 10",
"generated_cpe":"o:microsoft:windows_10:20h2:::::::",
"condition":"KB5005565 patch is not installed"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"low","authentication":"none","confidentiality_impact":"partial","integrity_impact":"none","availability":"none"},"base_score":"5"},"cvss3":{"vector":{"attack_vector":"network","access_complexity":"low","privileges_required":"none","user_interaction":"none","scope":"unchanged","confidentiality_impact":"high","integrity_impact":"none","availability":"none"},"base_score":"7.500000"}},
"cve":"CVE-2021-1678","title":"CVE-2021-1678 affects Windows 10","rationale":"NTLM Security Feature Bypass Vulnerability","severity":"High","published":"2021-01-12","updated":"2021-01-20","cwe_reference":"NVD-CWE-noinfo","
references":["https://portal.msrc.microsoft.com/en-US/security-guidance/advisory/CVE-2021-1678","https://nvd.nist.gov/vuln/detail/CVE-2021-1678"],"assigner":"secure@microsoft.com","cve_version":"4.0"}},"location":"vulnerability-detector"}
```

Number of vulnerabilities that the agent has, by executing the following API command:
`curl -k -X GET "https://localhost:55000/vulnerability/097" -H  "Authorization: Bearer $TOKEN"`
```
,"total_affected_items": 1,
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language
- [x] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [x] Scan-build report
  - [ ] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer

- [x] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [x] The data flow works as expected (agent-manager-api-app)

- New DisplayVersion field:
- [x] Windows agent without the field:
`sqlite3 /var/ossec/queue/db/098.db "select * from sys_osinfo" -header`
```
scan_id|scan_time|hostname|architecture|os_name|os_version|os_codename|os_major|os_minor|os_patch|os_build|os_platform|sysname|release|version|os_release|checksum|os_display_version
0|2021/09/20 12:17:31|WIN10-1909|x86_64|Microsoft Windows 10 Enterprise Evaluation|10.0.18363||10|0||18363|||||1909|1632140250300268200|
```
`Vulnerability detector logs`
```
2021/09/20 12:18:02 wazuh-modulesd:vulnerability-detector[114398] wm_vuln_detector.c:5767 at wm_vuldet_insert_agent_data(): DEBUG: (5446): The CPE 'o:microsoft:windows_10:1909:::::::' from the agent '098' was indexed.
```
- [x] Windows agent with collected field:
`sqlite3 /var/ossec/queue/db/097.db "select * from sys_osinfo" -header`
```
scan_id|scan_time|hostname|architecture|os_name|os_version|os_codename|os_major|os_minor|os_patch|os_build|os_platform|sysname|release|version|os_release|checksum|os_display_version
0|2021/09/20 11:33:16|WIN10|i686|Microsoft Windows 10 Home N|10.0.19042||10|0||19042|||||2009|1632137595599520900|20H2
```
`Vulnerability detector logs`
```
2021/09/20 12:12:57 wazuh-modulesd:vulnerability-detector[114398] wm_vuln_detector.c:5767 at wm_vuldet_insert_agent_data(): DEBUG: (5446): The CPE 'o:microsoft:windows_10:20h2:::::::' from the agent '097' was indexed.
```

### CODE REVIEW

File: src/data_provider/src/osinfo/sysOsInfoWin.cpp [hunk 1]
```diff
@@ -162,6 +162,17 @@ static std::string getRelease(const std::string& build)
...
 
+static std::string getDisplayVersion()
+{
+    std::string display_version;
+    if(IsWindows8OrGreater())
+    {
+        Utils::Registry currentVersion{HKEY_LOCAL_MACHINE, R"(SOFTWARE\Microsoft\Windows NT\CurrentVersion)"};
+        display_version = currentVersion.string("DisplayVersion");
+    }
+    return display_version;
```
Comment (chemamartinez) [inline src/data_provider/src/osinfo/sysOsInfoWin.cpp:173]: What happens if `display_version` is not filled? Please try it with an older Windows version to check its behavior.

Comment (MarcelKemp) [reply src/data_provider/src/osinfo/sysOsInfoWin.cpp:175]: When the `DisplayVersion` field did not exist, it caused an error when trying to collect it.

To fix this problem, I have uploaded a commit by first checking that the `DisplayVersion` field exists and if it doesn't, then assigning it an `unknown value`.

**Comment by chemamartinez** [comment]
Remember to merge the documentation PR after this one: https://github.com/wazuh/wazuh-documentation/pull/4377

**Review by chemamartinez — APPROVED** [review]
LGTM

**Comment by mmbar1986** [comment]
Hi everybody , 
thank you , but i can't understand what i have to do  
Can you explain more , i'am new here !!

Thank you so much

**Comment by MarcelKemp** [comment]
Hi @mmbar1986,

Sorry if I have not explained it correctly.

This PR includes the fix of the case that you have reported, and it will contain the new version **4.3** of Wazuh.

So when **_Wazuh v4.3_** comes out, you just need to **update** your environment to get the fixes and improvements of this PR.

**Comment by mmbar1986** [comment]
Hi  @MarcelKemp  , 

Great !! thank you !!
Best Regards

**Pull request merged**